File: src/ProjectR/content/scripts/spells/VengefulSpirit.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Vengeful Spirit"
end

function GetDescription()
  return "Send out a spirit to attack your target.\nLowers DEF and can inflict PNS."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .6
end

function GetDelay()
  return .32
end

function GetSpellType()
  return Composite
end

function GetMasteries()
  return { HOL }
end

function SpellEffect()
  local damage = ((2.625 * aAD + 2.625 * aMD) * (aHOL/100) - (.875 * dDEF + .875 * dMR)) * (100/dHOL)  
  d:BuffStat(DEF, -.5)
  d:ApplyDebuff(PSN, 35)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/VengefulSpirit.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class VengefulSpirit : SpellScriptBase
    {
        public override string Name { get { return "Vengeful Spirit"; } }
        public override string Description { get { return "Send out a spirit to attack your target.\nLowers DEF and can inflict PNS."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.HOL, }; } }
        public override SpellType SpellType { get { return SpellType.Composite; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 60; } }
        public override double Delay { get { return .32; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = ((2.625 * aAD + 2.625 * aMD) * (aHOL / 100) - (.875 * dDEF + .875 * dMR)) * (100 / dHOL);
            BuffStat(Stat.DEF, -.5);
            TryToApplyDebuff(DebuffResistance.PSN, 35);
            DealDamage(damage);
        }
    }
}